app/backend/src/couchers/servicers/threads.py

```diff
@@ -58,6 +59,18 @@ def GetThread(self, request, context):
                     for r, n in res[:page_size]
                 ]
 
+                # query without .limit() to get the total number of responses
+                all_res = (
+                    session.query(Comment, func.count(Reply.id))
+                    .outerjoin(Reply, Reply.comment_id == Comment.id)
+                    .filter(Comment.thread_id == database_id)
+                    .group_by(Comment.id)
+                    .all()
+                )
+                num_responses = 0
+                for r in all_res:
+                    num_responses += 1 + r[1]
```

Comment:
Please compute this within postgres, that'll speed it up a lot, and also you won't need to ship the actual data from the db. Something like

```py3
num_responses = (session.query(func.sum(1 + func.count(Reply.id))
  .outerjoin(Reply, Reply.comment_id == Comment.id)
  .filter(Comment.thread_id == database_id)
  .scalar())
```

This is completely untested and I'm not sure about the sum there, as well as the `.scalar()` (I'm pretty sure that's used somewhere in our codebase, `grep` for it, or just use `.one()[0]`)